pkg/api/addons.go

```diff
@@ -770,6 +772,50 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		},
 	}
 
+	defaultSecretsStoreCSIDriverAddonsConfig := KubernetesAddon{
+		Name:    common.SecretsStoreCSIDriverAddonName,
+		Enabled: to.BoolPtr(DefaultSecretStoreCSIDriverAddonEnabled && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.16.0")),
+		Containers: []KubernetesContainerSpec{
+			{
+				Name:           common.CSILivenessProbeContainerName,
+				Image:          specConfig.MCRKubernetesImageBase + k8sComponents[common.CSILivenessProbeContainerName],
+				CPURequests:    "10m",
+				MemoryRequests: "20Mi",
+				CPULimits:      "200m",
+				MemoryLimits:   "200Mi",
+			},
+			{
+				Name:           common.CSINodeDriverRegistrarContainerName,
+				Image:          specConfig.MCRKubernetesImageBase + k8sComponents[common.CSINodeDriverRegistrarContainerName],
+				CPURequests:    "10m",
+				MemoryRequests: "20Mi",
+				CPULimits:      "200m",
+				MemoryLimits:   "200Mi",
+			},
+			{
+				Name:           common.CSISecretsStoreDriverContainerName,
+				Image:          "docker.io/deislabs/secrets-store-csi:v0.0.8",
```

Comment:
we prob need a mcr for this at some point and make sure it's part of the vhd script

Reply:
Let's add the remaining three container images to `install-dependencies.sh` that aren't there already with this PR.

@jchauncey, do we have an easy path towards setting up an MCR pipeline that will effectively consume `secrets-store-csi` releases at the above deislabs container registry endpoint?

Reply:
The remaining 3 container images are already in `install_dependencies.sh`. All CSI components share these images. For the MCR pipeline there currently isn't one. I'll set it up.

Reply:
How is that image being built? Can we just add that to the list of csi pipelines we already have?

Reply:
Updated the PR with the mcr image for driver